src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -298,17 +296,20 @@ export class NotebookEditor implements INotebookEditor {
 
     public runAbove(cell: NotebookCell | undefined): void {
         if (cell && cell.index > 0) {
-            // Get all cellIds until `index`.
-            //const cells = this.document.cells.slice(0, cell.index);
-            const cells = this.document.getCells(new NotebookRange(0, cell.index));
-            this.runCellRange([...cells]);
+            void this.commandManager.executeCommand(
+                'notebook.cell.execute',
+                { start: 0, end: cell.index },
+                cell.notebook.uri
+            );
         }
     }
     public runCellAndBelow(cell: NotebookCell | undefined): void {
         if (cell && cell.index >= 0) {
-            // Get all cellIds starting from `index`.
-            const cells = this.document.getCells(new NotebookRange(cell.index, this.document.cellCount));
-            this.runCellRange([...cells]);
+            void this.commandManager.executeCommand(
+                'notebook.cell.execute',
+                { start: cell.index, end: cell.notebook.cellCount },
+                cell.notebook.uri
```

Comment:
This doesn't work, but easier to get this in now, waiting on a fix for https://github.com/microsoft/vscode/issues/123024